### PR TITLE
Add AsyncStateEvent helper class

### PR DESCRIPTION
**Pull request by emil-furniss**
Just a little helper class to help centralize how the `pending-state` event is created.

### CODE REVIEW

File: helpers/README.md [hunk 1]
```diff
@@ -120,3 +120,13 @@ import { getUniqueId } from '@brightspace-ui/core/helpers/uniqueId.js';
 // gets a unique indexed id (for lifetime of page)
 getUniqueId();
 ```
+
+## asyncStateEvent
```
Comment (dlockhart) [inline helpers/README.md]: Nit: heading should be capitalized to match class name

File: mixins/async-container/README.md [hunk 2]
```diff
@@ -30,14 +30,10 @@ class MyComponent extends AsyncContainerMixin(LitElement) {
...
-Async children may dispatch the promise-carrying `pending-state` event through a variety of methods, for example by calling `dispatchEvent` directly.
+Async children may dispatch the promise-carrying `pending-state` event through a variety of methods, for example by using the `AsyncStateEvent` helper class to create the event and then calling `dispatchEvent` directly.
 
 ```js
-this.dispatchEvent(new CustomEvent('pending-state', {
-  composed: true,
-  bubbles: true,
-  detail: { promise: pendingPromise }
-}));
+this.dispatchEvent(new AsyncStateEvent(pendingPromise));
```
Comment (dlockhart) [inline mixins/async-container/README.md]: Can we switch `run-async.js` to use this as well?

Comment (emil-furniss) [reply mixins/async-container/README.md]: Yes, just updated to do just that

Comment (dbatiste) [reply mixins/async-container/README.md]: If you need a test page to try this out, dialog has one - `dialog-async.html`.

**Pull request merged**